core/src/genesis.rs

```diff
@@ -260,13 +260,22 @@ pub struct RawGenesisBlock {
 }
 
 impl RawGenesisBlock {
+    const WARN_ON_GENESIS_GTE: u64 = 1024 * 1024 * 1024; // 1Gb
+
     /// Construct a genesis block from a `.json` file at the specified
     /// path-like object.
     ///
     /// # Errors
     /// If file not found or deserialization from file fails.
     pub fn from_path<P: AsRef<Path> + Debug>(path: P) -> Result<Self> {
         let file = File::open(&path).wrap_err(format!("Failed to open {:?}", &path))?;
+        let size = file
+            .metadata()
+            .wrap_err("Unable to access genesis file metadata")?
+            .len();
+        if size >= Self::WARN_ON_GENESIS_GTE {
```

Comment:
are we checking file size or deserialized block size?

Reply:
file size